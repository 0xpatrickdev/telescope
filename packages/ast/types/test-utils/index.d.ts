import { ProtoType } from '@osmonauts/types';
import { ProtoStore } from '@osmonauts/proto-parser';
import { AminoParseContext } from '../src/encoding/context';
export declare const expectCode: (ast: any) => void;
export declare const printCode: (ast: any) => void;
export declare const prepareContext: (store: ProtoStore, protoFile: string) => {
    context: AminoParseContext;
    root: import("@osmonauts/types").ProtoRoot;
    protos: ProtoType[];
};
