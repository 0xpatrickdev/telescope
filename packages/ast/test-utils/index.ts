import generate from '@babel/generator';
import { ProtoStore, getNestedProto, parseProto } from '@osmonauts/proto-parser'
import { ProtoType } from '../src/encoding/proto/types';
import { ParseContext } from '../src/encoding/context';

export const expectCode = (ast) => {
    expect(
        generate(ast).code
    ).toMatchSnapshot();
}
export const printCode = (ast) => {
    console.log(
        generate(ast).code
    );
}

export const prepareContext = (store: ProtoStore, protoFile: string) => {
    const ref = store.findProto(protoFile);
    const traversed = ref.traversed
    if (!traversed) {
        throw new Error('not traversed!');
    }
    const proto: Record<string, ProtoType> = getNestedProto(traversed);
    const protos: ProtoType[] = Object.values(proto).filter(
        proto => proto.name.startsWith('Msg')
            &&
            !proto.name.endsWith('Response')
            &&
            proto.name !== 'Msg'
    );

    const context = new ParseContext(ref, store);

    return {
        context,
        root: traversed,
        protos
    };
};
