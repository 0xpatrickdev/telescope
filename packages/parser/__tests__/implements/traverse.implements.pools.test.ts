import { getTestProtoStore } from '../../test-utils';

describe('PoolI', () => {
  const store = getTestProtoStore();
  // @ts-ignore
  store.options.prototypes.implementsAcceptsAny = true;
  store.traverseAll();
  const queryRef = store.findProto('osmosis/gamm/v1beta1/query.proto');
  const balancerRef = store.findProto('osmosis/gamm/pool-models/balancer/balancerPool.proto');
  const stableswapRef = store.findProto('osmosis/gamm/pool-models/stableswap/stableswap_pool.proto');

  it('createMap', () => {
    const result = store.getTypeUrlMap(
      queryRef
    );

    expect(result).toEqual({
      PoolI: [
        {
          ref: 'osmosis/gamm/pool-models/balancer/balancerPool.proto',
          types: [
            {
              typeUrl: '/osmosis.gamm.v1beta1.Pool',
              aminoType: "osmosis/gamm/pool",
              type: 'Pool',
              importAs: 'Pool1'
            }
          ]
        },
        {
          ref: 'osmosis/gamm/pool-models/stableswap/stableswap_pool.proto',
          types: [
            {
              typeUrl: '/osmosis.gamm.poolmodels.stableswap.v1beta1.Pool',
              // TODO this is WRONG!!!!!!!!!!
              aminoType: "osmosis/gamm/pool",
              type: 'Pool',
              importAs: 'Pool2'
            }
          ]
        }
      ]
    })
  });

  it('balancerRef', () => {
    expect(balancerRef.traversed?.implementsInterface).toEqual({
      PoolI: ['Pool']
    })
  });
  it('stableswapRef', () => {
    expect(stableswapRef.traversed?.implementsInterface).toEqual({
      PoolI: ['Pool']
    })
  });
  it('queryRef', () => {
    expect(queryRef.traversed?.acceptsInterface).toEqual({
      PoolI: [
        'QueryPoolResponse',
        'QueryPoolsResponse',
        'QueryPoolsWithFilterResponse'
      ]
    })
  });
});