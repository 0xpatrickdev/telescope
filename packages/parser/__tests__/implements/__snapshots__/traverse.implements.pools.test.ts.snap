// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoolI createMap 1`] = `
{
  "PoolI": [
    {
      "pkg": "osmosis.gamm.v1beta1",
      "ref": "osmosis/gamm/pool-models/balancer/balancerPool.proto",
      "types": [
        {
          "aminoType": "osmosis/gamm/pool",
          "importAs": "Pool1",
          "type": "Pool",
          "typeUrl": "/osmosis.gamm.v1beta1.Pool",
        },
      ],
    },
    {
      "pkg": "osmosis.gamm.poolmodels.stableswap.v1beta1",
      "ref": "osmosis/gamm/pool-models/stableswap/stableswap_pool.proto",
      "types": [
        {
          "aminoType": "osmosis/gamm/pool",
          "importAs": "Pool2",
          "type": "Pool",
          "typeUrl": "/osmosis.gamm.poolmodels.stableswap.v1beta1.Pool",
        },
      ],
    },
  ],
}
`;
